packages/backend/src/utils/env.ts

```diff
@@ -27,24 +27,36 @@ export class GeneratorNotFoundError extends Error {
 }
 
 class EnvUtil {
-  private readonly existingNpmPaths: string[];
+  private existingNpmPathsPromise: Promise<string[]>;
   private allInstalledGensMeta: Environment.LookupGeneratorMeta[];
 
   constructor() {
-    // improves lookup speed by 1.5 sec
-    this.existingNpmPaths = (() => {
-      const globalNpmPaths: string[] = this.createEnvInstance().getNpmPaths();
-      const userNpmPaths = homedir()
-        .split(path.sep)
-        .map((part, index, parts) => {
-          const resPath = path.join(...parts.slice(0, index + 1), "node_modules");
-          return isWin32 ? resPath : path.join(path.sep, resPath);
-        });
-      // uniq and existing only paths (global npm path is always added)
-      const paths: string[] = _.union(globalNpmPaths, userNpmPaths).filter((npmPath: string) => existsSync(npmPath));
-      paths.push(NpmCommand.getGlobalNodeModulesPath());
-      return _.uniq(paths);
-    })();
+    this.existingNpmPathsPromise = this.getExistingNpmPath();
+  }
+
+  private getEnvNpmPath(): Promise<string[]> {
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        // this operation takes up to 2 seconds
+        // it should be wrapped with setTimeout to provide promise like behaviour
+        resolve(this.createEnvInstance().getNpmPaths());
+      }, 5);
+    });
```

Comment:
please explain. why 5 milisec ? it seems tricky ...

Reply:
Even 1 mills is enough.
I need the setTimeout to work and create promise